engine/language-server/src/main/scala/org/enso/languageserver/vcsmanager/VcsManagerApi.scala

```diff
@@ -0,0 +1,83 @@
+package org.enso.languageserver.vcsmanager
+
+import org.enso.jsonrpc.{Error, HasParams, HasResult, Method, Unused}
+import org.enso.languageserver.filemanager.Path
+
+object VcsManagerApi {
```

Comment:
Shouldn't this whole VCS related functionality be in its own module? Something like `engine/language-server-vcs`? It looks like a quite independent sub-system with its own dependency needs. Separating it into an independent compilation  unit would prevent other parts of `language-server` code to use JGit API directly when it is supposed to be hidden.

Option1: Define the `VcsApi` in 'language-server` module without any dependencies on JGit. Implement it in the `language-server-vcsimpl` module that will have dependency on JGit as well as `language-server` module. Use [ServiceLoader](http://wiki.apidesign.org/wiki/ServiceLoader) to connect them together.

Option2: Define the `VcsApi` in a separate module `language-server-vcs`. Implement it in `language-server-vcsimpl` module that will have dependency on JGit as well as `language-server-vcs` module. Change the `language-server` module to depend on `language-server-vcs`, use its API. The API module would use [ServiceLoader](http://wiki.apidesign.org/wiki/ServiceLoader) again to find the right implementation during runtime.

Reply:
Seemed like a bit of an overkill at this point. Mostly because we don't use any other VCS implementation at this point. If we had more implementations, I would totally agree with you.
Similar thing can be said about other elements of the language server like file manager and all buffer-related stuff.

Reply:
> Similar thing can be said about other elements of the language server

Yes. There are many other violations, but when will we start fixing them if not when introducing completely new library? Or will we continue building such a _megalitic monolith_ forever?

Reply:
That sounds to me more like a regular refactoring action that could be applied to different subprojects of `language-server`. Mixing that work with project save done here, would make review rather difficult. 

So how about we schedule that work with @4e6 separately?

Reply:
Created https://www.pivotaltracker.com/story/show/183794226 as a follow up work.